Content.Server/Radio/EntitySystems/HeadsetSystem.cs

```diff
@@ -111,4 +138,49 @@ private void OnEmpPulse(EntityUid uid, HeadsetComponent component, ref EmpPulseE
             args.Disabled = true;
         }
     }
+
+    private void OnPrototypeReload(PrototypesReloadedEventArgs args)
+    {
+        CacheChannels();
+    }
+
+    private void OnGetVerbs(EntityUid uid, HeadsetComponent component, ref GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanInteract || args.Hands == null)
+            return;
+
+        if (!_interaction.InRangeUnobstructed(args.User, args.Target))
+            return;
+
+        if (!TryComp(uid, out EncryptionKeyHolderComponent? keyHolder))
+            return;
+
+        foreach ((var channel, var index) in keyHolder.Channels.Select(static (channel, index) => (channel, index)))
+        {
+            var name = _channels[channel].LocalizedName;
+
+            var toggled = component.SoundChannels.Contains(channel);
+
+            args.Verbs.Add(new()
+            {
+                Text = toggled ? $"[bold]{name}" : name,
+                Priority = index,
+                Category = VerbCategory.ToggleHeadsetSound,
+                Act = () => ToggleHeadsetSound((uid, component), channel, !toggled)
+            });
+        }
+    }
+
+    private void CacheChannels()
+    {
+        _channels = _prototype.EnumeratePrototypes<RadioChannelPrototype>().ToFrozenDictionary(prototype => prototype.ID);
```

Comment:
I feel like you should probably just get this inside the verb with `_prototype.Index<RadioChannelPrototype>()`. Doing this array this is more complicated than it needs to be!